crates/autopilot/src/domain/settlement/trade.rs

```diff
@@ -29,6 +33,48 @@ impl Trade {
             prices,
         }
     }
+
+    /// Surplus based on custom clearing prices returns the surplus after all
+    /// fees have been applied.
+    ///
+    /// Denominated in NATIVE token
+    pub fn native_surplus(&self, prices: &auction::Prices) -> Result<eth::Ether, Error> {
+        let surplus = self.surplus().ok_or(Error::Surplus(self.sell, self.buy))?;
+        let price = prices
+            .get(&surplus.token)
+            .ok_or(Error::MissingPrice(surplus.token))?;
+
+        Ok(price.in_eth(surplus.amount))
+    }
+
+    /// Surplus based on uniform clearing prices returns the surplus without any
+    /// fees applied.
+    ///
+    /// Denominated in SURPLUS token
+    #[allow(dead_code)]
+    fn surplus_before_fee(&self) -> Option<eth::Asset> {
```

Comment:
When will this dead code be needed? Can we add it when the time comes?

Reply:
Yes i can remove it, but have in mind that having two functions is the main reason to extract function `trade_surplus` as a standalone function.